pkg/pgmodel/migrations/sql/idempotent/base.sql

```diff
@@ -1486,35 +1516,67 @@ BEGIN
                     RETURN;
         END IF;
 
-        RETURN QUERY
-        SELECT
-            m.id,
-            m.metric_name,
-            m.table_name,
-            SCHEMA_CATALOG.get_metric_retention_period(m.metric_name) as retention_period,
-            (
-                SELECT _timescaledb_internal.to_interval(interval_length)
-                FROM _timescaledb_catalog.dimension d
-                WHERE d.hypertable_id = h.id
-                ORDER BY d.id ASC
-                LIMIT 1
-            ) as chunk_interval,
-            ARRAY(
-                SELECT key
-                FROM SCHEMA_CATALOG.label_key_position lkp
-                WHERE lkp.metric_name = m.metric_name
-                ORDER BY key) label_keys,
-            hi.total_size as size,
-            (1.0 - (pg_size_bytes(chs.compressed_total_bytes)::numeric / pg_size_bytes(chs.uncompressed_total_bytes)::numeric)) * 100 as compression_ratio,
-            chs.total_chunks,
-            chs.number_compressed_chunks as compressed_chunks
-        FROM SCHEMA_CATALOG.metric m
-        LEFT JOIN timescaledb_information.hypertable hi ON
-                    (hi.table_schema = 'SCHEMA_DATA' AND hi.table_name = m.table_name)
-        LEFT JOIN timescaledb_information.compressed_hypertable_stats chs ON
-                    (chs.hypertable_name = format('%I.%I', 'SCHEMA_DATA', m.table_name)::regclass)
-        LEFT JOIN _timescaledb_catalog.hypertable h ON
-                    (h.schema_name = 'SCHEMA_DATA' AND h.table_name = m.table_name);
+        IF SCHEMA_CATALOG.get_timescale_major_version() >= 2 THEN
+            RETURN QUERY
+            SELECT
+                m.id,
+                m.metric_name,
+                m.table_name,
+                SCHEMA_CATALOG.get_metric_retention_period(m.metric_name) as retention_period,
+                (
+                    SELECT _timescaledb_internal.to_interval(interval_length)
```

Comment:
can we use the new info view for this https://docs.timescale.com/v2.0/api#timescaledb_information-dimensions

Reply:
🤩